File: VB/Watermark/ASPxProcessedImageGallery.ascx.vb
Imports DevExpress.Web
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.IO
Imports System.Linq
Imports System.Web
Imports System.Web.UI

Namespace Watermark
    Partial Public Class ASPxProcessedImageGallery
        Inherits System.Web.UI.UserControl

        <PersistenceMode(PersistenceMode.InnerProperty), DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public ReadOnly Property ImageGallery() As ASPxImageGallery
            Get
                Return ASPxImageGallery
            End Get
        End Property
        <PersistenceMode(PersistenceMode.Attribute), DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Property ProcessedImageFolder() As String
        <PersistenceMode(PersistenceMode.Attribute), DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Property SourceImageFolder() As String
        <PersistenceMode(PersistenceMode.Attribute), DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Property WatermarkImageUrl() As String
        Protected Sub Page_Init(ByVal sender As Object, ByVal e As EventArgs)
            For Each item As String In Directory.GetFiles(Server.MapPath(SourceImageFolder))
                If Not File.Exists(SourceImageFolder & "\" & Path.GetFileName(item)) Then
                    ProcessImage(item)
                End If
            Next item
            ImageGallery.SettingsFolder.ImageSourceFolder = ProcessedImageFolder
        End Sub

        Private Sub ProcessImage(ByVal name As String)
            Using image As Image = System.Drawing.Image.FromFile(name)
            Using watermarkImage As Image = System.Drawing.Image.FromFile(Server.MapPath(WatermarkImageUrl))
            Using imageGraphics As Graphics = Graphics.FromImage(image)
            Using watermarkBrush As New TextureBrush(watermarkImage)
                Dim x As Integer = (image.Width - watermarkImage.Width)
                Dim y As Integer = (image.Height - watermarkImage.Height)
                watermarkBrush.TranslateTransform(x, y)
                imageGraphics.FillRectangle(watermarkBrush, New Rectangle(New Point(x, y), New Size(watermarkImage.Width + 1, watermarkImage.Height)))
                image.Save(Server.MapPath(ProcessedImageFolder & "\" & Path.GetFileName(name)))
            End Using
            End Using
            End Using
            End Using
        End Sub
    End Class
End Namespace

File: CS/Watermark/ASPxProcessedImageGallery.ascx.cs
using DevExpress.Web;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Web;
using System.Web.UI;

namespace Watermark
{
	public partial class ASPxProcessedImageGallery : System.Web.UI.UserControl
	{
		[PersistenceMode(PersistenceMode.InnerProperty)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public ASPxImageGallery ImageGallery {
			get {
				return ASPxImageGallery;
			}
		}
		[PersistenceMode(PersistenceMode.Attribute)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public string ProcessedImageFolder { get; set; }
		[PersistenceMode(PersistenceMode.Attribute)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public string SourceImageFolder { get; set; }
		[PersistenceMode(PersistenceMode.Attribute)]
		[DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
		public string WatermarkImageUrl { get; set; }
		protected void Page_Init(object sender, EventArgs e) {
			foreach (string item in Directory.GetFiles(Server.MapPath(SourceImageFolder))) {
				if (!File.Exists(SourceImageFolder + "\\" + Path.GetFileName(item)))
					ProcessImage(item);
			}
			ImageGallery.SettingsFolder.ImageSourceFolder = ProcessedImageFolder;
		}

		private void ProcessImage(string name) {
			using (Image image = Image.FromFile(name))
			using (Image watermarkImage = Image.FromFile(Server.MapPath(WatermarkImageUrl)))
			using (Graphics imageGraphics = Graphics.FromImage(image))
			using (TextureBrush watermarkBrush = new TextureBrush(watermarkImage)) {
				int x = (image.Width - watermarkImage.Width);
				int y = (image.Height - watermarkImage.Height);
				watermarkBrush.TranslateTransform(x, y);
				imageGraphics.FillRectangle(watermarkBrush, new Rectangle(new Point(x, y), new Size(watermarkImage.Width + 1, watermarkImage.Height)));
				image.Save(Server.MapPath(ProcessedImageFolder + "\\" + Path.GetFileName(name)));
			}
		}
	}
}